src/polycubed/src/port.cpp

```diff
@@ -138,12 +138,24 @@ void Port::send_packet_out(const std::vector<uint8_t> &packet, Direction directi
   c.send_packet_to_cube(module, port, packet);
 }
 
+/*
+* TC programs receive as value on the update_forwarding_table function a value
+* representing the virtual id of the next port and the index of the next module
+* to jump.
+* On the other hand, XDP programs receive as value the ifindex of the interface
+* that will be inserted into the devmap in order to use the bpf_redirect_map helper.
+*/
 void Port::connect(Port &p1, Node &iface) {
   std::lock_guard<std::mutex> guard(p1.port_mutex_);
   p1.peer_iface_ = &iface;
   // TODO: provide an accessor for forward chain?
   // Even smarter, should be forward chain a property of Port?
-  p1.parent_.update_forwarding_table(p1.index_, iface.get_index());
+
+  if (p1.parent_.get_type() == CubeType::TC) {
+    p1.parent_.update_forwarding_table(p1.index_, iface.get_index(), true);
```

Comment:
I think the last parameter should be `false`,  only `XDP` programs have `bpf_redirect_map` support.

Reply:
The last parameter tells if the connection is with a `netdev` or not.
To me, it is correct to pass the value `true` even for TC cubes but, in the current implementation they simply ignore that value and use the default implementation the `bpf_redirect_map` in the `Cube` class.
On the other hand, `XDP Cubes` extend that function to consider the last parameter writing the value in the `BPF_DEVMAP`.

What do you think?

Reply:
It would be more clear to me if we create a define such as "IS_NETDEV true", so that the prototype of the function remains as is, but the readability of the code looks better: reader knows immediately what this parameter is about when it reads "IS_NETDEV" as param of this function.